### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by voloshynamaryna11**

### CODE REVIEW

File: src/main/java/internet/market/web/filters/AuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        String url = req.getServletPath();
+        if (url.equals("/user/login") || url.equals("/user/registration")) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        if (userId == null || userService.get(userId) == null) {
```
Comment (Vosker) [inline src/main/java/internet/market/web/filters/AuthenticationFilter.java]: Тебе тут вторая часть не нужна. В чатике обьясняли.

**Review by Vosker — COMMENTED** [review]
Cool!

File: src/main/java/internet/market/web/filters/AuthenticationFilter.java (hunk 1 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/web/filters/AuthenticationFilter.java]: ok thanks

File: src/main/java/internet/market/controllers/LoginController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        try {
+            User user = authenticationService.login(login, password);
+            HttpSession session = req.getSession();
+            session.setAttribute("user_Id", user.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/market/controllers/LoginController.java]:
```suggestion
            session.setAttribute(USER_ID, user.getId());
```
let's make it constant as in filter

File: src/main/java/internet/market/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import internet.market.exceptions.AuthenticationException;
+import internet.market.lib.Inject;
+import internet.market.lib.Service;
+import internet.market.model.User;
+import internet.market.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    UserService userService;
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/market/security/AuthenticationServiceImpl.java]: access modifier

File: src/main/java/internet/market/security/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User user = userService.findByLogin(login)
+                .orElseThrow(() ->
+                        new AuthenticationException("Register first or enter correct login"));
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/market/security/AuthenticationServiceImpl.java]:
```suggestion
                        new AuthenticationException("Wrong password or login"));
```
let's not give any clue to malefactors)

File: src/main/java/internet/market/web/filters/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final String USER_ID = "user_Id";
+    private static final Injector injector = Injector.getInstance("internet.market");
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/market/web/filters/AuthenticationFilter.java]:
```suggestion
```

File: src/main/java/internet/market/web/filters/AuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        if (userId == null) {
+            resp.sendRedirect(req.getContextPath() + "/user/login");
+            return;
+        }
+        filterChain.doFilter(req, resp);
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/internet/market/web/filters/AuthenticationFilter.java]:
```suggestion
```

File: src/main/java/internet/market/controllers/LoginController.java (hunk 2 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/controllers/LoginController.java]: done

File: src/main/java/internet/market/security/AuthenticationServiceImpl.java (hunk 3 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/security/AuthenticationServiceImpl.java]: done

File: src/main/java/internet/market/security/AuthenticationServiceImpl.java (hunk 4 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/security/AuthenticationServiceImpl.java]: done

File: src/main/java/internet/market/web/filters/AuthenticationFilter.java (hunk 5 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/web/filters/AuthenticationFilter.java]: done

File: src/main/java/internet/market/web/filters/AuthenticationFilter.java (hunk 6 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/web/filters/AuthenticationFilter.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**